integration_tests/juniper_tests/src/codegen/impl_scalar.rs

```diff
@@ -0,0 +1,42 @@
+use fnv::FnvHashMap;
+#[cfg(test)]
+use juniper::{
+    self, parser::ScalarToken, DefaultScalarValue, GraphQLType, InputValue, ParseScalarResult,
+    ParseScalarValue, Value,
+};
+
+struct UserId(String);
+
+// TODO Trait that the macro handles, move to proper location! Naming?
+trait ParseCustomScalarValue<T, S = DefaultScalarValue> {
+    fn resolve(&self) -> Value;
+    fn from_input_value(value: &InputValue) -> Option<T>;
+    fn from_str<'a>(value: ScalarToken<'a>) -> ParseScalarResult<'a, S>;
+}
+
+#[juniper::graphql_scalar2(name = "MyCustomName", description = "My custom description")]
+impl ParseCustomScalarValue<UserId> for UserId {
```

Comment:
For the existing proc macros this isn't a *real* trait for the impl I think. So this can just be `GraphQLScalar`. See https://github.com/graphql-rust/juniper/blob/master/integration_tests/juniper_tests/src/codegen/impl_union.rs#L37 for an example...there is no `GraphQLUnion` trait.